activejob/lib/active_job/queue_adapter.rb

```diff
@@ -14,19 +14,23 @@ module ClassMethods
       # Specify the backend queue provider. The default queue adapter
       # is the :inline queue. See QueueAdapters for more
       # information.
-      def queue_adapter=(name_or_adapter)
+      def queue_adapter=(name_or_adapter_or_class)
         @@queue_adapter = \
-          case name_or_adapter
+          case name_or_adapter_or_class
           when Symbol, String
-            load_adapter(name_or_adapter)
+            load_adapter(name_or_adapter_or_class)
           else
```

Comment:
The `Class` instance may well be the adapter singleton, so we have to duck-type.